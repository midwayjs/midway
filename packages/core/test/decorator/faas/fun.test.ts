import {
  FUNC_KEY,
  Provide,
  getClassMetadata,
  listModule,
  ServerlessFunction,
  ServerlessTrigger,
  ServerlessTriggerType,
} from '../../../src';

@Provide()
class TestFun1 {
  @ServerlessFunction({
    functionName: 'abcde'
  })
  @ServerlessTrigger(ServerlessTriggerType.HTTP, { method: 'get', path: '/' })
  @ServerlessTrigger(ServerlessTriggerType.API_GATEWAY, { method: 'get', path: '/' })
  @ServerlessTrigger(ServerlessTriggerType.MTOP)
  @ServerlessTrigger(ServerlessTriggerType.SSR)
  @ServerlessTrigger(ServerlessTriggerType.HSF)
  @ServerlessTrigger(ServerlessTriggerType.EVENT)
  @ServerlessTrigger(ServerlessTriggerType.CDN)
  @ServerlessTrigger(ServerlessTriggerType.LOG, {
    source: '',
    project: '',
    log: ''
  })
  @ServerlessTrigger(ServerlessTriggerType.MQ, {
    topic: '',
    tags: '',
    region: '',
    strategy: 'BACKOFF_RETRY'
  })
  @ServerlessTrigger(ServerlessTriggerType.OS, {
    bucket: '',
    events: '',
    filter: {
      prefix: '',
      suffix: ''
    }
  })
  @ServerlessTrigger(ServerlessTriggerType.TIMER, {
    type: 'every',
    value: '5m',
  })
  invoke() {
  }
}

describe('/test/faas/fun.test.ts', () => {
  it('fun decorator should be ok', () => {
    const c = getClassMetadata(FUNC_KEY, TestFun1);
    expect(c).toMatchSnapshot();

    const m = listModule(FUNC_KEY);
    expect(m.length).toEqual(1);
  });
});
