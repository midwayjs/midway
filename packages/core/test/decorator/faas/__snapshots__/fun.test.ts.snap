// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/faas/fun.test.ts fun decorator should be ok 1`] = `
Array [
  Object {
    "metadata": Object {
      "type": "every",
      "value": "5m",
    },
    "methodName": "invoke",
    "type": "timer",
  },
  Object {
    "metadata": Object {
      "bucket": "",
      "events": "",
      "filter": Object {
        "prefix": "",
        "suffix": "",
      },
    },
    "methodName": "invoke",
    "type": "os",
  },
  Object {
    "metadata": Object {
      "region": "",
      "strategy": "BACKOFF_RETRY",
      "tags": "",
      "topic": "",
    },
    "methodName": "invoke",
    "type": "mq",
  },
  Object {
    "metadata": Object {
      "log": "",
      "project": "",
      "source": "",
    },
    "methodName": "invoke",
    "type": "log",
  },
  Object {
    "metadata": Object {},
    "methodName": "invoke",
    "type": "cdn",
  },
  Object {
    "metadata": Object {},
    "methodName": "invoke",
    "type": "event",
  },
  Object {
    "metadata": Object {},
    "methodName": "invoke",
    "type": "hsf",
  },
  Object {
    "metadata": Object {},
    "methodName": "invoke",
    "type": "mtop",
  },
  Object {
    "metadata": Object {
      "method": "get",
      "path": "/",
    },
    "methodName": "invoke",
    "type": "apigw",
  },
  Object {
    "metadata": Object {
      "method": "get",
      "path": "/",
    },
    "methodName": "invoke",
    "type": "http",
  },
]
`;
